core/app/models/spree/taxon.rb

```diff
@@ -35,6 +35,7 @@ class Taxon < Spree::Base
       content_type: { content_type: ["image/jpg", "image/jpeg", "image/png", "image/gif"] }
 
     # @note This method is meant to be overridden on a store by store basis.
+    # @deprecated - use Spree::Config.search_filter_presenter_class instead
     # @return [Array] filters that should be used for a taxon
     def applicable_filters
```

Comment:
Not sure what's the best way but probably it could be a good idea to change this method content to use the new class. If someone was using/overriding `taxon.applicable_filters` the interface would be the same and it should continue to work but under the hood it would start using the new `Spree::Core::Search::ProductFilter` which does exactly the same things.